lib/bootstrapper/src/main/java/uk/gov/hmcts/rse/ccd/lib/LibRunner.java

```diff
@@ -21,6 +23,24 @@
 
 public class LibRunner {
     public static void main(String[] args) throws Exception {
+        try {
+            doRun(args);
+        } catch (Exception e) {
+            System.out.println("*** cftlib failed to start ***");
+            System.out.println("This is a cftlib bug, please report it in the rse-devtools slack channel");
+            System.out.println("https://hmcts-reform.slack.com/archives/C033F1GDD6Z");
+            e.printStackTrace();
+
+            // Immediately terminate upon an unhandled error in the runner.
+            // This will ensure the JVM terminates even if we've started other threads.
+            Runtime.getRuntime().halt(-1);
```

Comment:
Are there any concerns here about shutting down the docker services?

Reply:
We never shut them down at the moment, we always leave them running however we exit.

Reply:
They get recreated on next boot if using the clean boot mode, otherwise they're left running to help boot time during local dev.